src/app/ReadClient.h

```diff
@@ -88,11 +96,14 @@ class ReadClient : public Messaging::ExchangeDelegate
 
     enum class ClientState
     {
-        Uninitialized = 0,     ///< The client has not been initialized
-        Initialized,           ///< The client has been initialized and is ready for a SendReadRequest
-        AwaitingInitialReport, ///< The client is waiting for initial report
+        Uninitialized = 0,         ///< The client has not been initialized
+        Initialized,               ///< The client has been initialized and is ready for a SendReadRequest
+        AwaitingInitialReport,     ///< The client is waiting for initial report
+        AwaitingSubscribeResponse, ///< The client is waiting for subscribe response
+        SubscriptionIdle,          ///< The client is maintaining subscription
     };
 
+    bool IsValidSubscription(uint64_t & aSubscriptionId) { return aSubscriptionId == mSubscriptionId && mRoleId == RoleId::Subscribe; }
```

Comment:
Why do we pass the argument by reference? 

I wonder if 'Valid' is the best name. I would expect this name to be 'validation' which would check something (like non-negative numbers or maybe not 0 or something like that).

Insteade this checks if the client is a subscription for the given id. How about

```cpp
bool IsClientForSubscription(uint64_t subscription);
```